experimental/devices/mfrc522/mfrc522.go

```diff
@@ -20,107 +20,81 @@ import (
 	"periph.io/x/periph/experimental/devices/mfrc522/commands"
 )
 
-// BlockAccess defines the block access bits.
-type BlockAccess byte
-
-// SectorTrailerAccess defines the sector trailing block access bits.
-type SectorTrailerAccess byte
-
-// Access bits.
-const (
-	AnyKeyRWID    BlockAccess = iota
-	RAB_WN_IN_DN              = 0x02 // Read (A|B), Write (None), Increment (None), Decrement(None)
-	RAB_WB_IN_DN              = 0x04
-	RAB_WB_IB_DAB             = 0x06
-	RAB_WN_IN_DAB             = 0x01
-	RB_WB_IN_DN               = 0x03
-	RB_WN_IN_DN               = 0x05
-	RN_WN_IN_DN               = 0x07
-
-	KeyA_RN_WA_BITS_RA_WN_KeyB_RA_WA        SectorTrailerAccess = iota
-	KeyA_RN_WN_BITS_RA_WN_KeyB_RA_WN                            = 0x02
-	KeyA_RN_WB_BITS_RAB_WN_KeyB_RN_WB                           = 0x04
-	KeyA_RN_WN_BITS_RAB_WN_KeyB_RN_WN                           = 0x06
-	KeyA_RN_WA_BITS_RA_WA_KeyB_RA_WA                            = 0x01
-	KeyA_RN_WB_BITS_RAB_WB_KeyB_RN_WB                           = 0x03
-	KeyA_RN_WN_BITS_RAB_WB_KeyB_RN_WN                           = 0x05
-	KeyA_RN_WN_BITS_RAB_WN_KeyB_RN_WN_EXTRA                     = 0x07
-)
-
-// BlocksAccess defines the access structure for first 3 blocks of the sector
-// and the access bits for the sector trail.
-type BlocksAccess struct {
-	B0, B1, B2 BlockAccess
-	B3         SectorTrailerAccess
+// Dev is an handle to an MFRC522 RFID reader.
+type Dev struct {
+	LowLevel         *commands.LowLevel
```

Comment:
One thing you can do is to embed the pointer, i.e. do:
```
  *commands.LowLevel
```

without a name. This will expose all its methods, including its own embedded sync.Mutex. That's what lepton does; https://github.com/google/periph/blob/master/devices/lepton/lepton.go#L110

It is very versatile for the users while keeping the godoc tight.

Reply:
This is also exposes the low-level details / methods into a high-level interface. So I'd rather keep those separated. The whole point of extracting that low-level part was to create a high-level interface with **fewer** methods exposed.